src/pages/tutorials/frontend/custom-checkout/add-shipping-carrier.md

```diff
@@ -66,24 +68,17 @@ ComponentRegistrar::register(
         "psr-4": {
             "Vendor\\CustomShipping\\": ""
         }
-    },
-    "version": "1.0.0"
+    }
 }
 ```
 
 ### Source code of `app/code/Vendor/CustomShipping/etc/module.xml`
 
 ```xml
 <?xml version="1.0"?>
-<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
-    <module name="Vendor_CustomShipping" >
-        <sequence>
-            <module name="Magento_Store"/>
-            <module name="Magento_Sales"/>
-            <module name="Magento_Quote"/>
-            <module name="Magento_SalesRule"/>
-        </sequence>
-    </module>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="Vendor_CustomShipping"/>
```

Comment:
You might want to keep the sequence but I can see why you removed it to keep it short and clean.

Reply:
I removed it because it simply isn't necessary, not as an effort to keep it short and clean.

Including it causes Magento to perform an unnecessary re-assessment of module load order in `app/code/etc/config.php` when we specify sequence.

This is a feature of Magento that I feel the community at broad lacks a good understanding of, probably because of practices that go back to the Magento 1 era where we did not have the capability to define dependencies separately from load order.

In Magento 2:

- _hard_ dependencies are to be stored in `composer.json` within the `require` block
- _soft_ dependencies are to be stored in `composer.json` within the `suggest` block
- load order preferences are stored in `etc/module.xml` via sequence declarations.

Just because a module is dependent on another, does not mean there is any requirement that it must be loaded first.  The scenarios I am aware of in Magento 2 that actually require explicit control over load order are the following:

- If there is _existing_ XML configurations in the other module which you want to _override_. It is then necessary to have that module be loaded first, so that when Magento consolidates the XML of all enabled modules into a combined result, our XML value was pulled in _after_ the other module, and hence our value wins over the value that was defined in the other module.
  - It is important to note here, that appending _new_ configuration to existing XML of another module, does _not_ require us to modify load order. For example if I want to add a new `<field>` inside a `<group>` node of an `etc/adminhtml/system.xml`, and the group is defined within another module, there is no need to specify load order. Magento will combine the final XML result of both modules, and my field will be successfully registered into the group defined by the other module.

- Another scenario that I have been told about, that have not verified myself, is when we need to define requireJS configurations that override the existing configuration of another module. Again, appending new configurations here, does not require strict control of load order, only overriding of existing.

Reply:
A final note here, I do my best to discourage Magento developers from performing unnecessary defining of `<sequence>` nodes not so much because of the minor changes to `app/etc/config.php` that it causes. Rather, in a large and long lived project with a growing number of modules, we may eventually run ourselves into a situation where Magento is unable to successfully determine a valid load order, as sequences may have become circular. When in fact (most) modules probably didn't need to define sequence at all.

Also a final note, it is valid to define sequence, without a hard dependency on the module (in other words, a sequence on soft dependency). In this scenario, it would be valid for the other module to be disabled (or not installed into `vendor/` at all), despite us defining a `<sequence>` stating it needs to load first. This was not possible in Magento 1 due to the inability to define module load order module dependencies separate from each other.

Reply:
Keep in mind that the audience for this type of doc is rather inexperienced unlike us seasoned developers. The intent of my comment was merely to keep it or a single reference to show a typical and expected development pattern. 

What you described is valid but hardly what someone of this level should be considering. My thought behind my comment was it's better to show real examples. It's completely fine to remove it.